src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -206,9 +272,20 @@ impl WorkunitStore {
   ///
   pub fn heavy_hitters(&self, k: usize) -> HashMap<String, Option<Duration>> {
     use petgraph::Direction;
-
     let now = SystemTime::now();
-    let inner = self.inner.lock();
+    let mut inner = self.heavy_hitters_data.inner.lock();
+
+    let receiver = self.heavy_hitters_data.started_workunits_rx.lock();
+
+    while let Ok(started) = receiver.try_recv() {
+      Self::add_started_workunit_to_store(started, &mut inner);
+    }
```

Comment:
I don't think it matters from a performance perspective, but it would probably be good to keep the scope of the lock acquisition narrower to minimize the chance of deadlock. So could put both of these lock acquisitions inside of blocks:
```
{
  let receiver = ...lock()
  while .. {
    ..
  }
}
```